src/node.rs

```diff
@@ -248,13 +260,19 @@ impl fmt::Display for StaticNode {
             Self::Bool(b) => write!(f, "{b}"),
             Self::I64(n) => write!(f, "{n}"),
             Self::U64(n) => write!(f, "{n}"),
+            #[cfg(not(feature = "ordered-float"))]
             Self::F64(n) => write!(f, "{n}"),
+            #[cfg(feature = "ordered-float")]
+            Self::F64(n) => write!(f, "{}", n.0),
             Self::I128(n) => write!(f, "{n}"),
             Self::U128(n) => write!(f, "{n}"),
         }
     }
 }
 
+#[cfg(feature = "ordered-float")]
```

Comment:
could we derive eq instead to ensure if we ever end up breaking the constraint, the compiler yells at us?

Reply:
Eq doesn't have any methods associated with it and according to [the doc](https://doc.rust-lang.org/std/cmp/trait.Eq.html) the extra (over PartialEq) reflexive constraint can't be checked by the compiler - so it's up to the implementer to "guarantee it".